mne/source_estimate.py

```diff
@@ -2639,14 +2639,64 @@ def morph_data_precomputed(subject_from, subject_to, stc_from, vertices_to,
     return stc_to
 
 
+def _spatio_temporal_src_connectivity_vol(src, n_times):
+    assert len(src) == 1  # not a mixed source space
+    from sklearn.feature_extraction import grid_to_graph
+    vertices = np.where(src[0]['inuse'])[0]
+    n_vertices = len(vertices)
+    data = (1 + np.arange(n_vertices))[:, np.newaxis]
+    stc_tmp = VolSourceEstimate(data, vertices, tmin=0., tstep=1.)
+    img = stc_tmp.as_volume(src, mri_resolution=False)
+    img_data = img.get_data()[:, :, :, 0]
+    img_data = img_data.T
```

Comment:
I suspect that these lines can be replaced by direct logic on the source space, rather than doing a temp STC, volume export, then data grab. Have you looked?

Reply:
(This would also increase the number of volume source spaces that can be used to those without MRI, like volume source space within a generic sphere, because `as_volume` requires MRI vol source space)

Reply:
no I have not looked. I will just put an XXX note for later thing.